src/docs/develop/cli.md

```diff
@@ -180,31 +219,31 @@ Our CLI is open source. Contribute to the development of the Railway CLI by open
+The Railway CLI recently underwent a rewrite to improve the user and developer experience.
+The biggest change is that the new v3 CLI will use the new [Public API](/reference/public-api/).
 
 The new CLI is written in Rust and is currently available via all release channels.
 
 ### What's New?
 
-- The new CLI is written in Rust and is much faster than the previous version
-- The new CLI has interactive search for services + projects to link to
-- The new CLI now displays information about your data in cleaner, more readable tables
-- The new CLI has additional flags such as `--json` to output data in JSON format for scripting
+- Written in Rust; much faster than previous versions
+- Interactive search for services + projects when linking
+- Tabular displays for cleaner data visualization
 - New commands:
+    - `railway service` for linking to a project's service
+    - `railway domain` for generating a service domain
 
-  - `railway service` helper command to link to a project's service
-  - `railway add` now allows you to provision multiple database types at once
-  - `railway domain` now will generate a service domain
-  - `railway logs` now opens a direct websocket connection to the logs API and will stream logs in real time
+### What's Improved?
 
-- Added the `--json` flag to output data in JSON format for scripting
-- CLI commands now allow you to explicitly specify the project and environment you want to use via flags.
+- Commands now allow you to specify the project and environment via flags
+- Additional flags such as `--json` to output data in JSON format for scripting
+- `railway add` now allows you to provision multiple database types at once
+- `railway logs` now opens a direct websocket connection to the logs API and will stream logs in real time
 
-### Deprecations
+### What's Deprecated?
 
-- Undocumented commands such as `railway protect` and `railway shell` will be removed. We are re-thinking how we want to handle these features and will be adding them back in the future
+- `railway build` has been removed
+- `railway protect` has been removed
 - `railway variables add` will no longer give you the option to add a secret. We are re-thinking how we want to handle the secret flow within the CLI now since the addition of [Shared Variables](/develop/variables#shared-variables)
-- `railway down` will be sunset as we are looking into better ways to handle deployments and scaling
-- `railway version` is being removed in favor of using `-V` or `--version` flags.
-- `railway build` is being removed in favor of using `railway variables` to output a local .env file (also the name was confusing)
+- `railway version` has been removed, in favor of using `-V` or `--version` flags
 - `railway init` will no longer allow you to deploy a new Project with a Template. We have since re-worked the flow and realized that the Template deploy flow is best served in the dashboard
```

Comment:
We may wanna eventually stage this for removal after the CLI regressions project is completed.

Reply:
Yep, I wanted to remove it entirely, but v3 isn't exactly "done" until regressions are taken care of. We can slate this for 23Q3.